spec/classes/mod/jk_spec.rb

```diff
@@ -67,7 +67,7 @@
     include_examples 'Debian 11'
 
     context 'with only required facts and default parameters' do
-      let(:facts) { super().merge('ipaddress' => default_ip) }
+      let(:facts) { super().merge(networking: { ip: default_ip }) }
```

Comment:
This breaks things and you need to do a deep merge. In VP we have a helper for this: `override_facts()`. https://github.com/voxpupuli/voxpupuli-test#fact-handling describes the reasoning. Ideally this would be easier (maybe something in rspec-puppet?), but your current approach remotes all other networking facts (like FQDN).

Reply:
No other facts are used but I see your point, will update the code to account